optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/factory/MoveIteratorFactory.java

```diff
@@ -49,7 +50,7 @@
      * @throws UnsupportedOperationException if only {@link #createRandomMoveIterator(ScoreDirector, Random)} is
      * supported
      */
-    Iterator<Move> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
+    Iterator<Move_> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
```

Comment:
I was expecting this to change into
`Iterator<? extends Move<Solution_>> createOriginalMoveIterator(...)` to avoid needing an extra type on the interface (which is a pain).

Would that work?